src/app/beer_garden/router.py

```diff
@@ -154,127 +183,182 @@ def forward(operation: Operation):
         UnknownGardenException: The specified target garden is unknown
     """
-    target_garden = None
-    for garden in gardens:
-        if garden.name == operation.target_garden_name:
-            target_garden = garden
-            break
-
-    if not target_garden:
+    try:
+        conn_type, conn_info = garden_connections[operation.target_garden_name]
+    except KeyError:
         raise UnknownGardenException(
             f"Unknown child garden {operation.target_garden_name}"
         )
 
-    if target_garden.connection_type is None:
-        target_garden = beer_garden.garden.get_garden(target_garden.name)
-        cache_gardens()
+    try:
+        if conn_type.casefold() == "http":
+            return _forward_http(operation, conn_info)
+        else:
+            raise RoutingRequestException(f"Unknown connection type {conn_type}")
+    except Exception as ex:
+        logger.exception(f"Error publishing to forward{ex}")
 
-    if target_garden.connection_type and target_garden.connection_type.casefold() in [
-        "http",
-        "https",
-    ]:
-        return _forward_http(operation, target_garden)
-    else:
-        raise RoutingRequestException(
-            f"Unknown connection type {target_garden.connection_type}"
-        )
 
+def setup_routing():
+    """Initialize the routing subsystem
+
+    This will load the cached child garden definitions and use them to populate the
+    two dictionaries that matter, garden_lookup and garden_connections.
 
-def cache_gardens():
-    for garden in beer_garden.garden.get_gardens():
-        add_garden(garden)
+    It will then query the database for all local systems and add those to the
+    dictionaries as well.
+    """
+    local_garden_name = config.get("garden.name")
 
+    # We do NOT want to load local garden information from the database as the local
+    # name could have changed
+    for garden in get_gardens():
+        if garden.name != local_garden_name:
+            if (
+                garden.connection_type is not None
+                and garden.connection_type.casefold() != "local"
+            ):
+                # add_garden(garden)
 
-def add_garden(garden: Garden):
-    # Mark all systems as reachable by this garden
-    for system_name in garden.systems:
-        garden_map[system_name] = garden
+                # Mark all systems as reachable by this garden
+                for system_name in garden.systems:
+                    garden_lookup[system_name] = garden.name
 
-    # Add to the garden listing
-    gardens.append(garden)
+                # Add to the connection lookup
+                garden_connections[garden.name] = (
+                    garden.connection_type,
+                    garden.connection_params,
+                )
+            else:
+                logger.warning(f"Adding garden with invalid connection info: {garden}")
 
+    # Now add the local systems
+    local_systems = get_systems(filter_params={"local": True})
 
```

Comment:
I am guessing you added this to the model in Brewtils? Btw, I like this approach more. It is easier to figure out what is not local.

Reply:
Yeah, I did. I'm not sure how much I like this - still very much a WIP

Reply:
Until you find something better, I am good with it.